tests/integration/test_fim/test_basic_usage/test_basic_usage_deferred_delete_folder.py

```diff
@@ -0,0 +1,127 @@
+conf_params = {'TEST_DIRECTORIES': directory_str, 'MODULE_NAME': __name__}
+p, m = generate_params(extra_params=conf_params, modes=['whodata'])
+configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
+
+
+# callback
+def callback_detect_delete_event(line):
+    msg = r'.*Sending FIM event: (.+)$'
+    match = re.match(msg, line)
+
+    try:
+        event = json.loads(match.group(1))
+        if (event['type'] == 'event' and
+                event['data']['type'] == 'deleted' and
+                'process_name' not in event['data']['audit']):
+            return event
+    except (AttributeError, JSONDecodeError, KeyError):
+        pass
+
+    return None
+
+# fixtures
+
+
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+# tests
+@pytest.mark.parametrize('folder, file_list, filetype, tags_to_apply', [
+    (testdir1, ['regular0', 'regular1', 'regular2'], REGULAR, {'ossec_conf'},),
+    (testdir2, ['regular0', 'regular1', 'regular2'], REGULAR, {'ossec_conf'},)
+])
+def test_deferred_delete_file(folder, file_list, filetype, tags_to_apply,
+                              get_configuration, configure_environment,
+                              restart_syscheckd, wait_for_initial_scan):
+    """
+    Check if syscheckd detects 'deleted' events from the files contained
+    in a folder that are deleted in a deferred manner.
+
+    We first run the command in order to find the confirmation character in the os,
+    after that we delete the files
+
+    The events generated must not contain the process_name parameter in order to guarantee
+    it's a 4659 event that generated it
+
+    Parameters
+    ----------
+    folder : str
+        Directory where the files will be created.
+    file_list : list
+        Names of the files.
+    filetype : str
+        Type of the files that will be created.
+    """
+    check_apply_test(tags_to_apply, get_configuration['tags'])
+
+    # Create files inside subdir folder
+    for file in file_list:
+        create_file(filetype, folder, file, content='')
+
+    # Wait for the added events
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout, callback=callback_detect_event,
+                            accum_results=len(file_list), error_message='Did not receive expected '
+                            '"Sending FIM event: ..." event')
+
+    # Delete the files under 'folder'
+    command = 'del "{}"\n'.format(folder)
+
+    cmd = Popen(command, shell=True, stdin=PIPE, stdout=PIPE, stderr=DEVNULL, universal_newlines=True)
+    try:
+        stdout = cmd.communicate(timeout=global_parameters.default_timeout)
+    except TimeoutError:
+        pass
+
+    # Find the windows confirmation character
+    confirmation = re.search(r'\((\w)\/\w\)\?', stdout[0])
+    assert confirmation
+
+    # Run the command again and this time delete the files

```

Comment:
Instead of `this time delete the files`, we could comment `confirm deletion of files`

Reply:
Done